### PR TITLE
Verify proof

### PR DESCRIPTION
**Pull request by calummoore**

### CODE REVIEW

**Comment by calummoore** [comment]
@mateuszmlc could you take a look at this? I'm getting `Error: constraint evaluations over the out-of-domain frame are inconsistent` from Miden when I try to verify the proof.

**Comment by mateuszmlc** [comment]
> @mateuszmlc could you take a look at this? I'm getting `Error: constraint evaluations over the out-of-domain frame are inconsistent` from Miden when I try to verify the proof.

Pushed a fix https://github.com/polybase/polylang/pull/121/commits/f4c84e129381685a1ea70f70c9c434107ea8518b

**Review by mateuszmlc — APPROVED** [review]

File: wasm-api/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      //
+      // the mode `development` makes `wasm-pack` build in `debug` mode.
+      // the mode `production` makes `wasm-pack` build in `release` mode.
+      // forceMode: "development",
+
+      // Controls plugin output verbosity, either 'info' or 'error'.
+      // Defaults to 'info'.
+      // pluginLogLevel: 'info'
+    }),
+    inlineWASM('pkg'),
```
[the remark this answers is not in the document]
Comment (calummoore) [reply wasm-api/webpack.config.js:97]: Is there a better way?

[the remark this answers is not in the document]
Comment (mateuszmlc) [reply wasm-api/webpack.config.js:97]:
```sh
$ cat pkg/wasm_playground_bg.wasm | gzip | wc -c
1754650

$ cat pkg/wasm_playground_bg.wasm | base64 | gzip | wc -c
2428046

$ cat pkg/wasm_playground_bg.wasm | hexdump -v -e '/1 "%02x"' | gzip | wc -c
2058792
```

Found this interesting, using a hex encoding with gzip gives a better result than using base64 with gzip, so inlining with hex would be more efficient, although still worse than just including the wasm as an asset and it being gzipped directly.

[the remark this answers is not in the document]
Comment (calummoore) [reply wasm-api/webpack.config.js:97]: Yeah, but if we package as a dependency, then it adds an extra step for everyone? What are we actually saving here like 1MB?

[the remark this answers is not in the document]
Comment (mateuszmlc) [reply wasm-api/webpack.config.js:97]: The base64 version will be 38% bigger for the end user to download + the browser can't streamingly compile it. But yeah, fair, the savings are not that big, plus if we want the package to be used by others, not just our frontend, then it makes more sense.

**Review by timmyjose — APPROVED** [review]

**Comment by sisou** [comment]
After this PR, https://polylang.dev/docs/getting-started/polylang-lib was not updated at the end of the `src/main.rs` code:

```diff
- file.write_all(&proof.to_bytes())?;
+ file.write_all(&proof.0.to_bytes())?;
```

**Pull request merged**